js/client.js

```diff
@@ -37,8 +37,52 @@ function getThreadClient() {
   return currentThreadClient;
 }
 
+const unsolicitedEvents = [
```

Comment:
Can you remove the unsolicitedEvents here? I don't think we should add them here. Otherwise this PR is fine to merge